cmd/nerdctl/compose_cp_linux_test.go

```diff
@@ -0,0 +1,72 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/containerd/nerdctl/pkg/rootlessutil"
+	"github.com/containerd/nerdctl/pkg/testutil"
+	"gotest.tools/v3/assert"
+)
+
+func TestComposeCopy(t *testing.T) {
+	if rootlessutil.IsRootless() {
+		t.Skip("Test skipped in rootless mode for compose copy")
```

Comment:
?

Reply:
HI @AkihiroSuda 

When I try to impletement the rooless mode, there are a liitle diffcult part to solve :-)

The command should use ` listContainers` , so that it need to run in the `rootless parent main`, 
But if it run in the  `rootless parent main`, it can not `nsenter` into the containerd to copy.

Error log in rootless mode:
```
ubuntu@ubuntu:~/oss/nerdctl$ ./_output/nerdctl --debug compose -f c.yaml cp /tmp/a svc0:/tmp
DEBU[0000] stateDir: /run/user/1000/containerd-rootless
DEBU[0000] rootless parent main: executing "/usr/bin/nsenter" with [-r/ -w/home/ubuntu/oss/nerdctl --preserve-credentials -m -n -U -t 12803 -F ./_output/nerdctl --debug compose -f c.yaml cp /tmp/a svc0:/tmp]
DEBU[0000] filters: [labels."com.docker.compose.project"==nerdctl,labels."com.docker.compose.service"==svc0]
INFO[0000] copy /tmp/a to nerdctl-svc0-1:/tmp
FATA[0000] error while executing [/home/ubuntu/oss/nerdctl/_output/nerdctl --debug=true --debug cp /tmp/a 16ac25a2abc1baaf1989c0e380ec998601152dbda19c884aab401bac382b310f:/tmp]: "time=\"2023-09-14T05:49:34Z\" level=debug msg=\"Detected tar binary \\\"/usr/bin/tar\\\" (GNU=true)\"\ntime=\"2023-09-14T05:49:34Z\" level=debug msg=\"executing [cp -a /tmp/a /tmp/nerdctl-cp874125715/a]\"\ntime=\"2023-09-14T05:49:34Z\" level=debug msg=\"executing [/usr/bin/tar -c -f - a] in \\\"/tmp/nerdctl-cp874125715\\\"\"\ntime=\"2023-09-14T05:49:34Z\" level=debug msg=\"executing [nsenter -t 15175 -U --preserve-credentials -- /usr/bin/tar -x -f -] in \\\"/proc/15175/root/tmp\\\"\"\nnsenter: reassociate to namespace 'ns/user' failed: Invalid argument\ntime=\"2023-09-14T05:49:34Z\" level=fatal msg=\"could not find container: 16ac25a2abc1baaf1989c0e380ec998601152dbda19c884aab401bac382b310f, with error: failed to wait [nsenter -t 15175 -U --preserve-credentials -- /usr/bin/tar -x -f -]: exit status 1\"\n": exit status 1
```

Could we mark it as a TODO, and reslove it later :-)

Reply:
SGTM, but please document that limitation

Reply:
Thanks @AkihiroSuda 
The comments has been added in the code and docs :-)

Reply:
- https://github.com/containerd/nerdctl/pull/2513